src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLLayer.m

```diff
@@ -122,7 +123,40 @@ - (void) blitTexture {
                 sourceSize:MTLSizeMake(src_w, src_h, 1)
                 toTexture:mtlDrawable.texture destinationSlice:0 destinationLevel:0 destinationOrigin:MTLOriginMake(0, 0, 0)];
         [blitEncoder endEncoding];
+#else
+        id<MTLCommandBuffer> cb = [ctx createCommandBuffer];
+        id<MTLComputeCommandEncoder> computeEncoder = [cb computeCommandEncoder];
+        id<MTLComputePipelineState> computePipelineState = [ctx.pipelineStateStorage
+                                                            getComputePipelineState:@"tex2tex_opaque"];
+        [computeEncoder setComputePipelineState:computePipelineState];
+
+        NSUInteger maxTotalThreadsPerThreadgroup = computePipelineState.maxTotalThreadsPerThreadgroup;
+        NSUInteger w = computePipelineState.threadExecutionWidth;
+
+        // Workaround for some OS/device bug reporting incorrect maxTotalThreadsPerThreadgroup
```

Comment:
@avu Do we know in which hardware we have this issue? Also do we have any reference to Apple bug?
Also if we use thread group as 1 what is the performance impact in such hardware's?

Reply:
Forced maxTotalThreadsPerThreadgroup to 1 and tested in Intel SoC 2015 Macbook Pro, i see almost ~50% reduction in FPS numbers in many use cases in RenderPerdTest. This final blit hits common operation and we would be able to make better decision whether this performance reduction is okay or not based on hardware in which we are seeing Apple bug.